test/unit/mongo/MongoTripodDriverTest.php

```diff
@@ -777,61 +848,612 @@ public function testSaveChangesWithInvalidCardinality()
+                            _ID_CONTEXT=>'http://talisaspire.com',
+                        ),
+                        OP_TABLES,
+                        'tripod_php_testing',
+                        'CBD_testing'
+                    )
+                )
+                ->setMethods(array('update'))
+                ->getMock(),
+            $this->getMockBuilder('\Tripod\Mongo\ImpactedSubject')
+                ->setConstructorArgs(
+                    array(
+                        array(
+                            _ID_RESOURCE=>'http://example.com/2',
+                            _ID_CONTEXT=>'http://talisaspire.com',
+                        ),
+                        OP_TABLES,
+                        'tripod_php_testing',
+                        'CBD_testing'
+                    )
+                )
+                ->setMethods(array('update'))
+                ->getMock()
+        );
+
+        $mockTables->expects($this->once())
+            ->method('getImpactedSubjects')
+            ->will($this->returnValue($impactedTableSubjects));
+
+        $impactedTableSubjects[0]->expects($this->once())->method('update');
+        $impactedTableSubjects[1]->expects($this->once())->method('update');
+
+        $mockTables->expects($this->never())->method('update');
+
+        $mockTripodUpdates->expects($this->once())
+            ->method('getDiscoverImpactedSubjects')
+            ->will($this->returnValue($mockDiscoverImpactedSubjects));
+
+        $mockTripodUpdates->expects($this->once())
+            ->method('storeChanges')
+            ->will($this->returnValue($subjectsAndPredicatesOfChange));
+
+        $mockTripod->expects($this->exactly(2))
+            ->method('getComposite')
+            ->will($this->returnValueMap(
+                array(
+                    array(OP_TABLES, $mockTables),
+                    array(OP_VIEWS, $mockViews)
+                )
+            ));
+
+        $mockTripod->expects($this->once())
+            ->method('getDataUpdater')
+            ->will($this->returnValue($mockTripodUpdates));
+
+        $mockDiscoverImpactedSubjects->expects($this->once())
+            ->method('createJob')
+            ->with(
+                $jobData,
+                \Tripod\Mongo\Config::getDiscoverQueueName()
+            );
+
+        $mockTripod->saveChanges($oG, new \Tripod\ExtendedGraph(),"http://talisaspire.com/");
+    }
+
+    public function testDiscoverImpactedSubjectsForDefaultOperationsSetting()
+    {
+        $uri_1 = "http://example.com/1";
+        $uri_2 = "http://example.com/2";
+        $oG = new \Tripod\Mongo\MongoGraph();
+        $oG->add_resource_triple($uri_1, $oG->qname_to_uri("rdf:type"), $oG->qname_to_uri("acorn:Resource"));
+        $oG->add_resource_triple($uri_2, $oG->qname_to_uri("rdf:type"), $oG->qname_to_uri("acorn:Resource"));
+
+        // a delete and an update
+        $nG = new \Tripod\Mongo\MongoGraph();
+        $nG->add_graph($oG);
+        // This should
```

Comment:
It might not even need a docblock - if so, there are a bunch of them lower down that we can amend/remove